src/AspNet.Security.OAuth.KakaoTalk/KakaoTalkAuthenticationOptions.cs

```diff
@@ -0,0 +1,57 @@
+﻿/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System;
+using System.Security.Claims;
+using System.Text.Json;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using static AspNet.Security.OAuth.KakaoTalk.KakaoTalkAuthenticationConstants;
+
+namespace AspNet.Security.OAuth.KakaoTalk
+{
+    /// <summary>
+    /// Defines a set of options used by <see cref="KakaoTalkAuthenticationHandler"/>.
+    /// </summary>
+    public class KakaoTalkAuthenticationOptions : OAuthOptions
+    {
+        public KakaoTalkAuthenticationOptions()
+        {
+            ClaimsIssuer = KakaoTalkAuthenticationDefaults.Issuer;
+            CallbackPath = KakaoTalkAuthenticationDefaults.CallbackPath;
+
+            AuthorizationEndpoint = KakaoTalkAuthenticationDefaults.AuthorizationEndpoint;
+            TokenEndpoint = KakaoTalkAuthenticationDefaults.TokenEndpoint;
+            UserInformationEndpoint = KakaoTalkAuthenticationDefaults.UserInformationEndpoint;
+
+            ClaimActions.MapJsonKey(ClaimTypes.NameIdentifier, "id");
+            ClaimActions.MapCustomJson(ClaimTypes.Name, user => GetValue(user, "kakao_account:profile:nickname"));
```

Comment:
This can be refactored, as it's the only custom action required, to just directly obtain this element.

Reply:
I didn't understand. Could you give me a little more specific explanation or provide examples or learning materials?

Reply:
Do you mean to call MapCustomJson with inline function as resolver?